jni/com_eclipsesource_v8_V8Impl.cpp

```diff
@@ -1469,3 +1472,215 @@ JNIEXPORT jlong JNICALL Java_com_eclipsesource_v8_V8__1getBuildID
   (JNIEnv *, jobject) {
   return 2;
 }
+
+/***** Serializes objects to JSON on the fly (used by code to serialize profiling information to file) *****/
+
+static void findAndReplace(string& source, const string find, const string replace)
+{
+    string::size_type i = 0;
+    while((i = source.find(find, i)) != string::npos) {
+      source.replace(i, find.length(), replace);
+      i += replace.length();
+    }
+}
+
+class JsonSerializer {
+  public:
```

Comment:
If you want, I can split out this class into separate header/cpp files. I left it currently in place as this prevents changes you would potentially have to make to the build process (as the g++ command would now have new files)